### PR TITLE
Fixes to DiSCO update logic

### PR DESCRIPTION
**Pull request by karenhanson**
The way isLatest() was being used caused DiSCOs never to update, partly because of the ambiguity of what HarvestRecordStatus.isLatest(RecordInfo) means - whether it means the record in the registry isLatest, or the new RecordInfo isLatest.
So, previously isLatest() was set to true when the harvested RecordInfo passed in was newer than the the version listed in the RecordRegistry.  Now, isLatest has been renamed to isUpToDate(), and it is true when the record in the registry isUpToDate.  If the registry indicates the record isUpToDate, the newly harvested RecordInfo does not need to be processed. New logic reflects this and DiSCO updates work as expected. An integration test that simulates a DiSCO being harvested 3 times demonstrates the behavior.
Also implemented a minor path correction for DiSCO updates, where the full API URL was recorded as the DiSCO URI, rather than just e.g. rmap:fjskj2dk
Fixed integration tests to reflect these changes.

### CODE REVIEW

File: rmap-loader-deposit-disco/src/main/java/info/rmapproject/loader/deposit/disco/DiscoDepositConsumer.java [hunk 1]
```diff
@@ -139,8 +138,7 @@ public void accept(HarvestRecord record) {
             if (response.getStatusLine().getStatusCode() == 201) {
                 harvestRegistry.register(
                         record.getRecordInfo(),
-                        URI.create(response.getFirstHeader("Location")
```
Comment (birkland) [inline rmap-loader-deposit-disco/src/main/java/info/rmapproject/loader/deposit/disco/DiscoDepositConsumer.java]: What's the motivation for this change?  Just by knowing HTTP/1.1, the client can presume that the Location header has a URI to the created resource; which is sufficient for the purposes of the harvester.  Is there a particular advantage to using the URI from the entity body?

Comment (karenhanson) [inline rmap-loader-deposit-disco/src/main/java/info/rmapproject/loader/deposit/disco/DiscoDepositConsumer.java]: This was causing an error because the getUpdateUri method was appending the Location URL to the disco endpoint.  This was resulting in something like this for the update path: https://rmap-hub.org/api/discos/https%3A%2F%2Frmap-hub.org%2Fapi%2F... etc.  I could either just set the location header as the update path, or save the actual DiSCO URI in the database. I chose the latter since there maybe other uses for the value in the URI field.

Comment (birkland) [inline rmap-loader-deposit-disco/src/main/java/info/rmapproject/loader/deposit/disco/DiscoDepositConsumer.java]: Ah, I see.  At some point, it would be great if it were possible to deploy an RMap for integration test purposes (backed by an in-memory triple store, or something), so that these things can be tested against the actual API.

In any case, what's in this PR looks fine!

Comment (karenhanson) [inline rmap-loader-deposit-disco/src/main/java/info/rmapproject/loader/deposit/disco/DiscoDepositConsumer.java]: Funnily enough, this is what @emetsger has been working on for the last few weeks - you can spin up a working test environment, it stores the test data until you close it down when you're done.  https://github.com/rmap-project/rmap/blob/master/DEVELOPER.md
